server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStats.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore.stats;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.remote.RemoteRefreshSegmentTracker;
+
+import java.io.IOException;
+
+/**
+ * Encapsulates all remote store stats
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreStats implements Writeable, ToXContentFragment {
+
+    private final RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats;
+
+    public RemoteStoreStats(RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats) {
+        this.remoteSegmentUploadShardStats = remoteSegmentUploadShardStats;
+    }
+
+    public RemoteStoreStats(StreamInput in) throws IOException {
+        remoteSegmentUploadShardStats = in.readOptionalWriteable(RemoteRefreshSegmentTracker.Stats::new);
+    }
+
+    public RemoteRefreshSegmentTracker.Stats getStats() {
+        return remoteSegmentUploadShardStats;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject()
+            .field("shardId", remoteSegmentUploadShardStats.shardId)
+            .field("latest_upload_files_count", remoteSegmentUploadShardStats.latestUploadFilesCount)
+            .field("latest_local_files_count", remoteSegmentUploadShardStats.latestLocalFilesCount)
+            .field("local_refresh_time", remoteSegmentUploadShardStats.localRefreshTime)
```

Comment:
If this provides `timestamp`, maybe we need to change the name of the metric accordingly?

Reply:
sure. this provides nanos as well i think. will update